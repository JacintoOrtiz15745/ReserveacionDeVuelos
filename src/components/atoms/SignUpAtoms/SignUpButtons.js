import React, {useEffect, useState} from 'react';
import {View, Text} from 'react-native';
import {styles} from '../../../styles/SignUpStyles';
import GoogleSignUp from './GoogleSignUp';
import {SignUpConstants} from '../../../utils/Constants';
import SignUpButton from './SignUpButton';
import {signUpSubmit} from '../../../utils/Functions';
import LoginMessage from './LoginMessage';
const SignUpButtons = ({
  validationValues,
  formValues,
  validateInputs,
  navigation,
}) => {
  const [isButtonEnabled, setIsButtonEnabled] = useState(false);
  useEffect(() => {
    const res = validateInputs(validationValues);
    setIsButtonEnabled(res);
  }, [validationValues]);
  return (
    <View style={styles.btnSignupContainer} testID="view">
      <SignUpButton
        isButtonEnabled={isButtonEnabled}
        formValues={formValues}
        signUpSubmit={signUpSubmit}
      />
      <Text style={styles.or}>{SignUpConstants.OR}</Text>
      <GoogleSignUp />
      <LoginMessage navigation={navigation} />
    </View>
  );
};

export default SignUpButtons;
