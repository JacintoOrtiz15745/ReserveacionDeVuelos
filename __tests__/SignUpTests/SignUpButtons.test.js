import React from 'react';
import SignUpButtons from '../../src/components/atoms/SignUpAtoms/SignUpButtons';
import {shallow} from 'enzyme';

const formValues = {
  name: '',
  email: '',
  password: '',
  termsCheckbox: false,
  updatesCheckbox: false,
  validName: true,
  validEmail: true,
  validPassword: true,
  validatedName: false,
  validatedEmail: false,
  validatedPassword: false,
};
const validationValues = {
  terms: formValues.termsCheckbox,
  updates: formValues.updatesCheckbox,
  validatedName: formValues.validatedName,
  validatedEmail: formValues.validatedEmail,
  validatedPassword: formValues.validatedPassword,
};
const validateInputsMock = jest.fn();
describe('renders correctly', () => {
  const wrapper = shallow(
    <SignUpButtons
      validationValues={validationValues}
      formValues={formValues}
      validateInputs={validateInputsMock}
    />,
  );
  test('snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('renders correctly', () => {
    const render = wrapper.findWhere((node) => node.prop('testID') === 'view');
    expect(render.exists()).toBeTruthy();
  });
  test('validate input func to have been called on mounted', () => {
    validateInputsMock();
    expect(validateInputsMock).toHaveBeenCalled();
  });
});
