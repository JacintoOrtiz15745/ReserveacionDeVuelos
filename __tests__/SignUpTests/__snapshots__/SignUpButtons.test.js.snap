// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
  testID="view"
>
  <Connect(SignUpButton)
    formValues={
      Object {
        "email": "",
        "name": "",
        "password": "",
        "termsCheckbox": false,
        "updatesCheckbox": false,
        "validEmail": true,
        "validName": true,
        "validPassword": true,
        "validatedEmail": false,
        "validatedName": false,
        "validatedPassword": false,
      }
    }
    isButtonEnabled={false}
  />
  <Text
    style={
      Object {
        "color": "#b6b7ba",
        "marginBottom": 10,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    or
  </Text>
  <Connect(GoogleSignUp) />
  <LoginMessage />
</View>
`;
